### ISSUE TITLE
[NBug] No application is associated with the specified file for th...

### ISSUE THREAD

**Issue by cyberluke** [gitextensions/gitextensions#7156]
:warning: The sections below must be filled in and this text must be removed or the issue will be closed.

## Error Details
```
System.ComponentModel.Win32Exception (0x80004005): No application is associated with the specified file for this operation
   at System.Diagnostics.Process.StartWithShellExecuteEx(ProcessStartInfo startInfo)
   at System.Diagnostics.Process.Start(ProcessStartInfo startInfo)
   at GitUI.CommandsDialogs.RevisionDiffControl.<>c.<diffOpenRevisionFileToolStripMenuItem_Click>b__53_0(String fileName)
   at GitUI.CommandsDialogs.RevisionDiffControl.<>c__DisplayClass55_0.<<SaveSelectedItemToTempFile>b__0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at GitUI.ThreadHelper.<>c__DisplayClass13_0.<<FileAndForget>b__0>d.MoveNext()
```

## Environment
- Git Extensions 3.1.1.6049
- Build 2f872105b4eb42421d9fdafd68e90a9e27d20d96
- Git 2.22.0.windows.1
- Microsoft Windows NT 10.0.18362.0
- .NET Framework 4.8.4010.0
- DPI 240dpi (250% scaling)

**Comment by gerhardol** [comment]
Seem to be new issue
"Open this revision (temp file)" for an extension without default way of opening. (Like .xlf for me)
Should have try-catch using "Open this revision with... (temp file)" as a backup.

**Comment by wischi-chr** [comment]
I can't reproduce this.

Created a new repo and added a file with a made up extension ".jjkp" (later also triex d ".xlf" just to be sure) and the option "Open this revision (temp file)" just opened the regular windows 10 application selector.

![image](https://user-images.githubusercontent.com/2495113/96774946-170c7e80-13e7-11eb-980f-40addf980df8.png)

Dialog text from the screenshot is in german, but I think you get the idea.
I gave the source a quick look and there is still no try catch - has the behaviour of Process.Start changed? 😮

**Comment by gerhardol** [comment]
> I gave the source a quick look and there is still no try catch - has the behaviour of Process.Start changed? 😮

The popup you got is the expected from Windows.
There are some hints in the other linked issues too, the problem occurs when the registered application is not available.
I tried to recreate agian as well, failed too.
try-catch or just use the modified exception reporting in #8358 or #8278

**Comment by mstv** [comment]
Please holler back to reopen if not resolved with 3.5.
